### ISSUE TITLE
Empty string returned when model.predict() for seq2seq model. 

### ISSUE THREAD

**Issue by nikhilbyte** [ThilinaRajapakse/simpletransformers#1326]
**Describe the bug**
I fine tuned a pre-trained model using the generic encoder decoder seq2seq minimal code mentioned in the docs. I trained the model for 2 Epochs and the loss converged to 0 in two epochs. When I try to evaluate the model, it evaluates properly with a loss of 0.0015.. but when I use the same model for inference it return empty strings as output.

**Screenshots**
<img width="1440" alt="Screenshot 2021-12-21 at 12 00 56 PM" src="https://user-images.githubusercontent.com/62005061/146883001-b4fc5aea-f429-4e9f-b9c8-a998df883390.png">
<img width="1078" alt="Screenshot 2021-12-21 at 12 03 37 PM" src="https://user-images.githubusercontent.com/62005061/146883044-7f5f476b-b17a-4a93-97aa-f2e9665a7021.png">
<img width="1078" alt="Screenshot 2021-12-21 at 12 03 47 PM" src="https://user-images.githubusercontent.com/62005061/146883061-35f9b9f8-0ed8-4ce6-9a43-e062318b914f.png">
<img width="1078" alt="Screenshot 2021-12-21 at 12 04 02 PM" src="https://user-images.githubusercontent.com/62005061/146883079-dfae1e20-3b08-4df6-a162-05fc9fbedd6d.png">
<img width="1078" alt="Screenshot 2021-12-21 at 12 04 19 PM" src="https://user-images.githubusercontent.com/62005061/146883090-53df5b88-9a26-42e3-a9bb-00322a447300.png">

**Comment by bingxin3chen** [comment]
Have you solved this problem？I also encountered this problem, and I don't know where the problem is.

**Comment by mapmeld** [comment]
Possibly related - I have trained seq2seq models with SimpleTransformers before, but on new models I now get this error that the Transformers library has changed:
https://github.com/huggingface/transformers/blob/master/src/transformers/models/encoder_decoder/modeling_encoder_decoder.py#L531

```
    "Version v4.12.0 introduces a better way to train encoder-decoder models by computing the loss inside the "
    "encoder-decoder framework rather than in the decoder itself. You may observe training discrepancies if fine-tuning "
    "a model trained with versions anterior to 4.12.0. The decoder_input_ids are now created based on the labels, no "
    "need to pass them yourself anymore."
```

**Comment by xuehaouwa** [comment]
> @bingxin3chen No, I switched to huggingface :)

Hi @nikhilbyte  Any examples/tutorials of training a similar Seq2Seq in huggingface?

Thanks

**Comment by nikhilbyte** [comment]
HI @xuehaouwa , I couldn't find a direct implementation like this but I implemented it on my own using huggingface.

**Comment by xuehaouwa** [comment]
@nikhilbyte  Would you please share your implementation with me? I have been struggling with this and I am new to NLP :(

It would be a great help if you could share a basic script. Thank you very much!
